### ISSUE TITLE
Left 4 Dead 2 does not launch on linux

### ISSUE THREAD

**Issue by MartinGggggggg** [ValveSoftware/Source-1-Games#6135]
Directly after clicking "play", the game is stuck forever when is says "Loading", but there is ever no progress at all.
Also tried using launch options as: -novid -vulkan, but it makes no difference.

![l4d2_20240707_124605](https://github.com/ValveSoftware/Source-1-Games/assets/170895916/713079b4-c3fe-44bf-bc7d-099e38435882)

OS: openSUSE Slowroll
Kernel: 6.9.7-1-default
GPU: Radeon 7600
CPU: AMD Ryzen 5 5500

Running the native Steam client from the official SUSE repo

**Comment by MartinGggggggg** [comment]
[l4d2log.txt](https://github.com/user-attachments/files/16179590/l4d2log.txt)
Forgot to attach a log

**Comment by Tiagoquix** [comment]
Are you using custom mods/add-ons?
Are you testing the game in a clean state?
Try launching with `-autoconfig -default`.

**Comment by MartinGggggggg** [comment]
> Are you using custom mods/add-ons? Are you testing the game in a clean state? Try launching with `-autoconfig -default`.

No, I have not added anything myself, just a clean installation from Steam.

But anyway, I just removed the installation completely and downloaded again. Made a verification of the integrity of game files and tried again.
Now, it crashes instead of stalling just before the picture attached earlier.
The suggested options does not make any difference.

**Comment by NULLYUKI** [comment]
Is it possible for you to share a crash log?

**Comment by MartinGggggggg** [comment]
> Is it possible for you to share a crash log?

Yes. I am not sure howto generare a crashlog, but I added "2>&1 | tee ~/l4d2_output.txt" as a launch option and captured the output into a file. Is this what you mean or is there another log with more info?
[l4d2_output.txt](https://github.com/user-attachments/files/16192693/l4d2_output.txt)

**Comment by NULLYUKI** [comment]
I think the provided log is already good, but I'm not a Linux expert, so maybe some other users will later provide some more information if there is some other way, to gain much more information. But the current provided log should be helpful.

**Comment by MartinGggggggg** [comment]
I also started the game with strace and captured the output if that gives any more information
[l4d2_strace.txt](https://github.com/user-attachments/files/16193280/l4d2_strace.txt)

**Comment by Tiagoquix** [comment]
Can you test on an older kernel, such as 6.8?

**Comment by MartinGggggggg** [comment]
> Can you test on an older kernel, such as 6.8?

I just tried 6.6.37-1-longterm, but the result is the same.

**Comment by MartinGggggggg** [comment]
I have also tried to run different versions of the steam linux runtime to see if one work better, but no luck.
Now I get  this message when I try to start "Unable to load /path_my_my_steam/steamapps/common/Left 4 Deam 2/bin/filesystem_steam.so"
There is no such file in the installation

**Comment by MartinGggggggg** [comment]
I have tried to run the game through Proton too. The problem then is that the connection is blocked blocked from accessing the server.
Can you remove this block so the game can be run in compatibility mode on linux if the native linux version is too hard to fix?

![l4d2_20240707_135940](https://github.com/user-attachments/assets/88e13ab0-ac46-4bf2-9fef-f5a2a490a6dc)

**Comment by Tiagoquix** [comment]
Source Engine games do not allow Proton versions to play on VAC-secured servers. You have to play with the native Linux version if you want to play online.

**Comment by Nshtk** [comment]
Bump, native version with `-vulkan` does not work at all (crashes without opening any windows), facing the same issue on 
```
- Arch
- Hyprland (Wayland)
- Nvidia driver 580.76.05
- Steam from official repo
```
Tried various launch options like: ```SDL_VIDEODRIVER=wayland STEAM_COMPAT_RUNTIME_SDL3=1 %command% -heapsize 750000 -vulkan -novid```

**Comment by Nshtk** [comment]
### Solution

Looks like the current version of Left 4 Dead 2 is not compatible with SDL3.
So to launch the game you need to set `STEAM_COMPAT_RUNTIME_SDL2=1`

**Comment by Nshtk** [comment]
### Solution №2
If you're also experiencing crashes on loading screen, try to use this launch option: `-malloc=system`
